src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,329 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using Iot.Device.Ssd1351.Command;
+
+namespace Iot.Device.Ssd1351
+{
+    /// <summary>
+    /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+    /// light emitting diode dot-matrix graphic display system. 
+    /// </summary>
+    public partial class Ssd1351 : IDisposable
+    {
+
+        private readonly SpiDevice _spiDevice;
+        private readonly GpioController _gpioDevice;
+        private readonly int _dcPinId;
+        private readonly int _resetPinId;
+        private ColorDepth _colorDepth;
+        private ColorSequence _colorSequence;
+        private int _spiBufferSize;
+
+        private const byte SCREEN_WIDTH_PX = 128;
+        private const byte SCREEN_HEIGHT_PX = 128;
+        private const int DefaultSPIBufferSize = 0x1000;
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">>The SPI device used for communication.</param>
+        /// <param name="gpioDevice">>The GPIO controller device used for communication.</param>
+        /// <param name="dcPinId">>The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPinId">>The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        public Ssd1351(SpiDevice spiDevice, GpioController gpioDevice, int dcPinId, int resetPinId)
```

Comment:
>gpioDevice [](start = 59, length = 10)

Why do you need a gpio controller to get passed in? usually what we do in other bindings, is that we initialize this gpio controller within the class and then dispose it once we dispose the object.

Reply:
I have made this modification. I initially felt that as I was passing in the SPI device then passing in the GPIO controller would be correct but I take your lead.